source/client-side-encryption/client-side-encryption.md

```diff
@@ -472,6 +473,35 @@ See
 <span id="GCPKMSOptions"></span> <span id="AWSKMSOptions"></span> <span id="KMSProvider"></span>
 <span id="KMSProviders"></span> <span id="AzureAccessToken"></span> <span id="kmsproviders"></span>
 
+#### credentialProviders
```

Comment:
Can we add some tests for FLE?  I can think of 3 tests that might be worth adding:

1. A CE / auto encryption configured with explicit KMS credentials and custom KMS providers throws an error.
2. Tests that demonstrate the explicit encryption / auto encryption fetch credentials using a custom credential provider.
3. A test that demonstrates that, given a client configured for auto AWS credential fetching with a custom credential provider, auto encryption / explicit encryption can be configured with a separate AWS credential provider. 

I'm not actually sure that 3. is a valuable feature (is there a use case for this?) but it should be supported with the current spec, so maybe we should add a test for it to make it explicit.

Reply:
Just a note I've just added 1 & 2 as 3 became difficult to set up with our existing tooling and would require a significant amount to additions in tools for an edge case. Perhaps we can revisit later if we see a need.